### PR TITLE
Fixup PR-1679

### PR DESCRIPTION
**Pull request by adamretter**
Enhances and fixes some shortcoming of #1679

Closes #1679

### CODE REVIEW

**Comment by adamretter** [comment]
This has a small API modification around initialising DBBroker from BrokerPool. However, as I don't think anyone would ever directly initialise a DBBroker instance (i.e. they are always taken from BrokerPool) we should be okay to do this before a 5.0.0. Arguments anyone?

File: src/org/exist/storage/DBBroker.java [hunk 1]
```diff
@@ -123,11 +124,27 @@ public DBBroker(final BrokerPool pool, final Configuration config) {
...
+         * has been called since the previous lease of this broker.
+         */
+        loadIndexModules();
+    }
+
+    /**
+     * Loads the index modules via an IndexController
+     */
+    protected void loadIndexModules() {
+        indexController.setIfExpiredOrNull(getBrokerPool().getIndexManager().getConfigurationTimestamp(), () -> new IndexController(this));
```
Comment (shabanovd) [inline src/org/exist/storage/DBBroker.java]: this looks as over-engineering to me because broker have `single thread` contract

**Comment by dizzzz** [comment]
I agree, @wolfgangmm ? @ljo ?

**Comment by adamretter** [comment]
@dizzzz @shabanovd Are we good to merge now?

**Pull request merged**